Added balance (combat) to drinfomon

UFTimmy opened the pull request:
Useful if you need to regain balance in combat.

rcuhljr commented on drinfomon.lic:
```diff
@@ -699,6 +708,9 @@ while line = script.gets
       DRStats.circle = Regexp.last_match(1).to_i
+    elsif line =~ /^\[You're(.*\,)?(.*) (and|with).*\]$/
+      balance_map = ['completely imbalanced', 'hopelessly unbalanced', 'extremely imbalanced', 'very badly balanced', 'badly balanced', 'somewhat off balance', 'off balance', 'slightly off balance', 'solidly balanced', 'nimbly balanced', 'adeptly balanced', 'incredibly balanced']
+      DRStats.balance = balance_map.index(Regexp.last_match(2).strip) + 1
```
Most of the appraisal scales go from 0->#, any reason to start balance at 1 instead of zero? Also I think there might be an alternate messaging form when you kill an enemy where you only get balance and no positional element, probably a pretty niche concern.

UFTimmy commented:
I prefer zero index, but I somehow thought everyone else liked 1 indexing.

Changed it to be zero indexed, and updated the regex to account for the balance message when not facing anything. Thanks for catching that.
